specs/_features/maxeb_increase/capella.py

```diff
@@ -3540,24 +3751,41 @@ def has_eth1_withdrawal_credential(validator: Validator) -> bool:
     return validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
 
 
+def has_compounding_withdrawal_credential(validator: Validator) -> bool:
+    """
+    Check if ``validator`` has an 0x02 prefixed "compounding" withdrawal credential.
+    """
+    return validator.withdrawal_credentials[:1] == COMPOUNDING_WITHDRAWAL_PREFIX
+
+
 def is_fully_withdrawable_validator(validator: Validator, balance: Gwei, epoch: Epoch) -> bool:
     """
     Check if ``validator`` is fully withdrawable.
     """
     return (
-        has_eth1_withdrawal_credential(validator)
+        (has_eth1_withdrawal_credential(validator) or has_compounding_withdrawal_credential(validator))
         and validator.withdrawable_epoch <= epoch
         and balance > 0
     )
 
+def get_validator_excess_balance(validator: Validator, balance: Gwei) -> Gwei:
+    """
+    Get excess balance for partial withdrawals for ``validator``.
+    """
+    if has_eth1_withdrawal_credential(validator) and balance > MIN_ACTIVATION_BALANCE:
+        return balance - MIN_ACTIVATION_BALANCE
+    if has_compounding_withdrawal_credential(validator) and balance > MAX_EFFECTIVE_BALANCE:
+        return balance - MAX_EFFECTIVE_BALANCE
+    return Gwei(0)
+
 
 def is_partially_withdrawable_validator(validator: Validator, balance: Gwei) -> bool:
     """
     Check if ``validator`` is partially withdrawable.
     """
-    has_max_effective_balance = validator.effective_balance == MAX_EFFECTIVE_BALANCE
```

Comment:
It looks like the check `has_max_effective_balance` was removed by mistake.

There may be a situation where a validator has lost 0.25 eth below `MAX_EFFECTIVE_BALANCE` resulting in its effective balance decreasing by 1 eth. The validator could then bring the balance up to `MAX_EFFECTIVE_BALANCE` or higher, but less than `MAX_EFFECTIVE_BALANCE + 0.25 eth` (the value needed to return the effective balance to `MAX_EFFECTIVE_BALANCE`). 

The current code allows partial withdrawals of the validator balance when `effective balance < MAX_EFFECTIVE_BALANCE`, because of which it is very likely that the validator balance will not reach the necessary `MAX_EFFECTIVE_BALANCE + 0.25 eth` to return its effective balance to equal `MAX_EFFECTIVE_BALANCE`.

The same is true for 0x01 type and `MIN_ACTIVATION_BALANCE`

Reply:
@avsetsin Please can you port your review comments to https://github.com/ethereum/consensus-specs/pull/3618 ?